deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/graph/ComputationGraph.java

```diff
@@ -644,15 +646,17 @@ public void init(INDArray parameters, boolean cloneParametersArray) {
                 continue; //Output vertex
             VertexIndices[] outputIndices = new VertexIndices[thisVertexOutputsTo.size()];
             int j = 0;
-            for (String s : thisVertexOutputsTo) {
+            for (String s : new HashSet<>(thisVertexOutputsTo)) {
```

Comment:
I don't think we need this new hashset construction here... we're not modifying thisVertexOutputsTo in the loop, so it should be fine as-is?

Reply:
thisVertexOutputsTo contains duplicates that we need in order to calculate the size of outputIndices correctly. But once we start iterating over it we are only interested in the unique values. Creating a HashSet was just the smallest change that I could think of.